### PR TITLE
Fix for AllowPrivilegeEscalation : issue-1403

### PR DESCRIPTION
**Pull request by neelasha-09**
Fix for #1403

### CODE REVIEW

**Comment by neelasha-09** [comment]
@FxKu @sdudoladov  - This is the fix for the bug introduced in OPR 1.6 that does not allow `cron` to work any longer.
A quick review would be highly appreciated. We tested and everything is working fine again with no errors. 
Spilo Issue: https://github.com/zalando/spilo/issues/562 which we reviewed with @CyberDem0n

**Comment by FxKu** [comment]
Thanks for your contribution @neelasha-09 . Looks good. The name for the parameter is fine. Two minor things.
1. In yaml manifests I'd like to keep alphabetical sorting so move `spilo_allow_privilege_escalation` further up. 
2. I'm thinking if the default should be `false`, too, like with privileged mode. OTOH looking at the use cases you've mentioned (cron, cert rotation) it seems to be more user-friendly if it's set to true straight away. Other opinions?

**Comment by neelasha-09** [comment]
> Thanks for your contribution @neelasha-09 . Looks good. The name for the parameter is fine. Two minor things.
> 
> 1. In yaml manifests I'd like to keep alphabetical sorting so move `spilo_allow_privilege_escalation` further up.
We thought of keeping the `spilo_allow_privilege_escalation` and `spilo_privileged`  next to each other as they are closely related.
> 2. I'm thinking if the default should be `false`, too, like with privileged mode. OTOH looking at the use cases you've mentioned (cron, cert rotation) it seems to be more user-friendly if it's set to true straight away. Other opinions?
The default value to `false` would cause BWC with OPR  version1.5.0. 
Without this parameter, functionality would be decreased. In our opinion, it should not be postponed  till next milestone & rather consider as fix within same release.

**Comment by FxKu** [comment]
1. Makes sense, but would also apply to many other existing options. Alphabetical sorting makes it easier in the end to find stuff. And with the `spilo_` prefix the options are not that far away from each other anyway.
2. Good point keeping it consistent with 1.5.0.

**Comment by neelasha-09** [comment]
> 1. Makes sense, but would also apply to many other existing options. Alphabetical sorting makes it easier in the end to find stuff. And with the `spilo_` prefix the options are not that far away from each other anyway.

                 Sorted the parameters in alphabetical order as suggested. 
> 2. Good point keeping it consistent with 1.5.0.

                 Also, could you please consider of not postponing the fix until 1.7.0  and fix is within the same release

File: pkg/controller/operator_config.go [hunk 1]
```diff
@@ -68,6 +68,7 @@ func (c *Controller) importConfigurationFromCRD(fromCRD *acidv1.OperatorConfigur
 	result.PodEnvironmentSecret = fromCRD.Kubernetes.PodEnvironmentSecret
 	result.PodTerminateGracePeriod = util.CoalesceDuration(time.Duration(fromCRD.Kubernetes.PodTerminateGracePeriod), "5m")
 	result.SpiloPrivileged = fromCRD.Kubernetes.SpiloPrivileged
+	result.SpiloAllowPrivilegeEscalation =fromCRD.Kubernetes.SpiloAllowPrivilegeEscalation
```
Comment (FxKu) [inline pkg/controller/operator_config.go:71]:
```suggestion
	result.SpiloAllowPrivilegeEscalation = fromCRD.Kubernetes.SpiloAllowPrivilegeEscalation
```

File: docs/reference/operator_parameters.md [hunk 2]
```diff
@@ -374,6 +374,11 @@ configuration they are grouped under the `kubernetes` key.
   used for AWS volume resizing and not required if you don't need that
   capability. The default is `false`.
 
+* **spilo_allow_privilege_escalation**
+  Controls whether a process can gain more privileges than its parent
+  process. Required by cron which needs setuid. Without this parameter, certification
+  rotation & backups will not be done. The default is `true`.
```
Comment (FxKu) [inline docs/reference/operator_parameters.md:380]:
```suggestion
  Controls whether a process can gain more privileges than its parent
  process. Required by cron which needs setuid. Without this parameter,
  certification rotation & backups will not be done. The default is `true`.
```

**Comment by FxKu** [comment]
:+1:

**Comment by sdudoladov** [comment]
:+1:

**Comment by FxKu** [comment]
Thanks @neelasha-09 . We are making it a 1.6.2 release now. We cannot replace previous images/releases.

**Comment by neelasha-09** [comment]
Thanks @FxKu for the PR merge 
However, I've raised another PR with the correction in the default values for the parameter  `SpiloAllowPrivilegeEscalation` under `pkg/util/config/config.go` -  https://github.com/zalando/postgres-operator/pull/1434

**Pull request merged**